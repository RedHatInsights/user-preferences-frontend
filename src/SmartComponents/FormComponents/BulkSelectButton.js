import React from 'react';
import { Button, Tooltip } from '@patternfly/react-core';
import PropTypes from 'prop-types';
import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
import './BulkSelectButton.scss';
import { useSearchParams } from 'react-router-dom';

const BulkSelectButton = (props) => {
  const formOptions = useFormApi();
  const { input, section, onClick } = useFieldApi({
    ...props,
    type: 'button',
  });

  const [searchParams] = useSearchParams();

  const showTooltipAndDisableButton =
    searchParams.get('bundle') === 'openshift' &&
    searchParams.get('app') === 'cluster-manager';

  return showTooltipAndDisableButton ? (
    <Tooltip content="You can not unsubscribe from these notifications. Please contact your org admin for more information.">
      <Button
        className="pref-c-bulk-select-button"
        variant="secondary"
        isDisabled={
          searchParams.get('bundle') === 'openshift' &&
          searchParams.get('app') === 'cluster-manager'
        }
        {...input}
        id={`bulk-select-${section}`}
        onClick={() => onClick?.(formOptions, input)}
      >
        {input.value ? 'Select' : 'Deselect'} all
      </Button>
    </Tooltip>
  ) : (
    <Button
      className="pref-c-bulk-select-button pf-v6-u-mb-md"
      variant="secondary"
      isDisabled={showTooltipAndDisableButton}
      {...input}
      id={`bulk-select-${section}`}
      onClick={() => onClick?.(formOptions, input)}
    >
      {input.value ? 'Select' : 'Deselect'} all
    </Button>
  );
};

BulkSelectButton.propTypes = {
  section: PropTypes.string,
  onClick: PropTypes.func,
};

export default BulkSelectButton;
