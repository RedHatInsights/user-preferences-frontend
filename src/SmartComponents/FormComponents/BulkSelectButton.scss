.pref-c-bulk-select-button {
    width: fit-content;
}