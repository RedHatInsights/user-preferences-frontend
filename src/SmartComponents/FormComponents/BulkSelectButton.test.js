import React from 'react';
import { Form, RendererContext } from '@data-driven-forms/react-form-renderer';
import { fireEvent, getByRole, render } from '@testing-library/react';
import BulkSelectButton from './BulkSelectButton';

const mockInputOnChange = jest.fn();

jest.mock('@data-driven-forms/react-form-renderer/use-field-api', () =>
  jest.fn((props) => ({
    ...props,
    input: {
      onChange: mockInputOnChange,
      value: true,
    },
  }))
);

describe('BulkSelectButton tests', () => {
  const onChangeMock = jest.fn();

  afterEach(() => {
    mockInputOnChange.mockReset();
    onChangeMock.mockReset();
  });

  it('should render correctly', () => {
    const { container } = render(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{
              formOptions: {
                internalRegisterField: jest.fn(),
                internalUnRegisterField: jest.fn(),
                getState: () => ({
                  dirtyFields: [],
                }),
              },
            }}
          >
            <BulkSelectButton label="My button" {...props} />
          </RendererContext.Provider>
        )}
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  it('should toggle correctly', () => {
    const { container } = render(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{
              formOptions: {
                internalRegisterField: jest.fn(),
                internalUnRegisterField: jest.fn(),
                batch: (callback) => callback(),
                getRegisteredFields: () => ['rhel[advisor]', 'rhel[sources]'],
                change: onChangeMock,
              },
            }}
          >
            <BulkSelectButton
              label="My button"
              group="rhel"
              section="sources"
              {...props}
            />
          </RendererContext.Provider>
        )}
      </Form>
    );

    fireEvent.click(getByRole(container, 'button'));
    expect(onChangeMock).toHaveBeenCalledWith('rhel[sources]', true);
    expect(mockInputOnChange).toHaveBeenCalledWith(false);
  });
});
