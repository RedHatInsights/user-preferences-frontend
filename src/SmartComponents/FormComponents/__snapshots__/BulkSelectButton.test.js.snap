// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkSelectButton tests should render correctly 1`] = `
<div>
  <button
    aria-disabled="false"
    class="pf-v5-c-button pf-m-secondary pref-c-bulk-select-button pf-v5-u-mb-md"
    data-ouia-component-id="OUIA-Generated-Button-secondary-1"
    data-ouia-component-type="PF5/Button"
    data-ouia-safe="true"
    id="bulk-select-undefined"
    type="button"
    value="true"
  >
    Select
     all
  </button>
</div>
`;
